src/codegen/properties/attribute.py

```diff
@@ -0,0 +1,281 @@
+        """
+        return self._is_object_type
+
+    @property
+    def c_function_name(self):
+        """
+        The name of the c function to be called when using the attribute.
+        This name will be prefixed with `DAQmxSet', `DAQmxGet` and `DAQmxReset` for using in getters, setters and deleters respectively.
+        """
+        return self._c_function_name
+
+    @property
+    def calling_convention(self):
+        """
+        The calling convention to be followed when using the c functions.
+        Possible values
+        - 'StdCall'
+        - 'Cdecl'
+        """
+        return self._calling_convention
+
+    @property
+    def is_list(self):
+        """
+        'True': If the attribute is a list.
+        'False': Id the attribute is a scalar value.
+        """
+        return self._is_list
+
+    @property
+    def bitfield_enum(self):
+        """
+        The name of the bitfield enum that the attribute represents
+        The value is `N/A` when it is not a bitField Enum.
+        During code generation in python, this will be used to decide if the `enum_to_bitfield_list` method needs to called in the getter when returning the value.
+        """
+        if(self._bitfield_enum == "N/A"):
+            return None
+        return self._bitfield_enum
+
+    @property
+    def has_explicit_read_buffer_size(self):
+        """
+        Specifies if an explicit read buffer size has to be provided when making the c function calls for the attribute.
+        If True then an additional uint parameter would be provided when calling the c function to mention the buffer size.
+        """
+        return self._has_explicit_read_buffer_size
+
+    @property
+    def read_buffer_size(self):
+        """
+        The read buffer size to be used when calling the c function.
+        This key would only be applicable if `has_explicit_read_buffer_size` is `True`.
+        In case the `has_explicit_read_buffer_size` is `True` and this key is not present, then the ivi dance method is used to get the buffer size.
+        """
+        if not hasattr(self,"_read_buffer_size"):
+            return None
+        return self._read_buffer_size
+
+    @property
+    def description(self):
+        """
+        The description of the attribute.
+        This will be used to define the docstring of the attribute when generating the code.
+        """
+        return self._description
+
+    @property
+    def python_data_type(self):
+        """
+        The python data_type of the attribute.
+        Currently this is used in the generation of the doc string for the attribute.
+        """
+        return self._python_data_type
+
+    @property
+    def ctypes_data_type(self):
+        """
+        The type of the attribute as per the ctypes definition in python.
+        This is used to provide the type of the attribute when making c function calls in python.
+        """
+        return self._ctypes_data_type
+
+class Parameter():
```

Comment:
Why is `Parameter` in the `attribute` module?

Reply:
Moved `Parameter` to a separate module